test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestApi.java

```diff
@@ -98,20 +99,20 @@ void addPath(String path) {
         this.paths.add(path);
     }
 
-    public List<String> getPathParts() {
+    public Map<String, Boolean> getPathParts() {
```

Comment:
Probably should have javadoc because it isn't obvious what the boolean reads unless you read it as part of this PR.